pkg/k8s/apis/cilium.io/v2/types.go

```diff
@@ -458,3 +459,43 @@ func (n *CiliumNode) InstanceID() (instanceID string) {
 	}
 	return
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:singular="ciliumegressgatewaypolicy",path="ciliumegressgatewaypolicies",scope="Cluster"
+// +kubebuilder:printcolumn:JSONPath=".metadata.labels.io\\.kubernetes\\.pod\\.namespace",description="The namespace of the entity",name="Namespace",type=string
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",description="The age of the identity",name="Age",type=date
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+
+type CiliumEgressGatewayPolicy struct {
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec CiliumEgressGatewayPolicySpec `json:"spec,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=false
+// +deepequal-gen=false
+
+// CiliumEgressGatewayPolicyList is a list of EgressGatewayPolicy objects.
+type CiliumEgressGatewayPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	// Items is a list of CiliumEgressGatewayPolicy
+	Items []CiliumEgressGatewayPolicy `json:"items"`
+}
+
+type CiliumEgressGatewayPolicySpec struct {
+	PodSelector       *slimv1.LabelSelector `json:"podSelector,omitempty"`
+	NamespaceSelector *slimv1.LabelSelector `json:"namespaceSelector,omitempty"`
```

Comment:
I'm not sure that the "slim" fields are meant to be used inside a CRD, but rather when setting up informers to watch for resources and those resources will instead of "slim" fields, so that we subject the apiserver to extraneous load. 

@aanm Is that right? Should "slim" fields be used in the definition of the CRD?

Reply:
> @aanm Is that right? Should "slim" fields be used in the definition of the CRD?

No, it should not for now.

Reply:
Thanks for the comment.

When I tried to use the metav1 definition, it seems to have some issues if DeepEqual:
```
/home/blzhao/go/src/github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2/zz_generated.deepequal.go:562:28: in.NamespaceSelector.DeepEqual undefined (type *"k8s.io/apimachinery/pkg/apis/meta/v1".LabelSelector has no field or method DeepEqual)
```
Similarly the slim definition is used for the cilium network policy:
https://github.com/cilium/cilium/blob/3a803bfcc4e7775ca5bb1daeede2c3c85f40969e/pkg/k8s/slim/k8s/api/networking/v1/types.go#L173

Reply:
@MasterZ40 ups sorry, I misread which slim fields Chris was referring to. Yes please use the `slim_metav1` similarly to the CiliumNetworkPolicy.

Reply:
Cool. Thx.
Resolving

Reply:
Just for my understanding, are the slim_meta fields the only ones "allowed" to be in CRDs?